miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/context/PrometheusCollectionFilter.java

```diff
@@ -0,0 +1,246 @@
+package uk.ac.bbsrc.tgac.miso.webapp.context;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Locale;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.springframework.web.filter.GenericFilterBean;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Histogram;
+import uk.ac.bbsrc.tgac.miso.core.util.CoverageIgnore;
+
+/**
+ * Collect per-URL performance statistics and export them to Prometheus
+ */
+@SuppressWarnings("deprecation")
```

Comment:
A bunch of methods and types in the `HttpServletResponse` are deprecated, but I can't avoid them to use the interface. If I put the suppression on just that class, it still whines about the imports.